src/app/coin/[id]/page.tsx

```diff
@@ -0,0 +1,308 @@
+"use client";
```

Comment:
This page has lot of logic. I would recommend not writing all JSX in once, instead move JSX to different different components and then load all of them here one below each other.